File: java/LiquidPlannerDemo.java
import java.util.List;
import java.util.HashMap;
import java.io.Console;

public class LiquidPlannerDemo {
  public static void main(String[] args) {  
  
    String email = null;
    Console console = System.console();
    
    System.out.print("Enter your e-mail address: ");
    email = console.readLine();
    System.out.print("Enter your password: ");
    char[] password = console.readPassword();
    LiquidPlanner LP = new LiquidPlanner(email, new String(password));
    LP.workspaces();
    System.out.println("Defaulting to your first workspace...");

    LP.set_workspace_id(((Double)LP.workspaces().get(0).get("id")).intValue());
    List<HashMap> projects = LP.projects();
    
    System.out.println("Here is a list of all your projects...");
    Integer project_id = ((Double)projects.get(0).get("id")).intValue();
    for (HashMap project : projects) {
      System.out.println(project.get("name"));
    }
    System.out.println("Adding a task...");
    LP.createTask("{\"name\":\"learn the API\",\"parent_id\":" + project_id.toString() + "}");
		
    System.out.println("Here is a list of all your tasks: ");
    List<HashMap> tasks = LP.tasks();
    
    for (HashMap<String, Object> task : tasks) {
      System.out.println(task.get("name"));
    }

  }
}

File: javascript/liquidPlannerDemo.js
var LiquidPlanner = require('./liquidPlanner'),
    read          = require('read'),
    flow          = require('flow');

var user, password, lp;

flow.exec(
  function getEmailAndPassword() { var done = this;

    read({ prompt: 'Enter your e-mail address: ' }, function(er, _email) {
      user = _email;

      read({ prompt: 'Enter your password: ', silent: true }, function(er, _password) {
        password = _password;
        
        done();
      });
    });

  },
  function connectToLiquidPlanner() {

    lp = new LiquidPlanner( user, password );

    this();

  },
  function workspaces() { var done = this;

    console.log("Defaulting to your first workspace...");
    lp.workspaces(function( workspaces ) {
      lp.setSpace(workspaces[0]);
      done(); 
    });

  }, 
  function showProjects() { var done = this;

    console.log("Here is a list of all your projects...");
    lp.projects(function( projects ) {
      for (var i = 0; i < projects.length; i++) {
        console.log(projects[i].name);
      }

      done();
    });

  }, 
  function addATask() {

    console.log("Adding a task...");
    lp.createTask({ name: "learn the API" }, this); 

  }, 
  function listAllTasks() { var done = this;

    console.log("Here is a list of all your tasks: ");
    lp.tasks(function(tasks) {
      for (var i = 0; i < tasks.length; i++) {
        console.log(tasks[i].name);
      }

      done();
    });

  }
);
